### PR TITLE
Minor vehicle performance optimizations (2019-07-24)

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: Performance "Minor vehicle performance optimizations"```

#### Purpose of change

Minor performance gain when waiting while there is a vehicle with automatic turrets nearby.

#### Describe the solution

Get rid of expensive translation calls in debug messages and do not use translated name when it probably won't be used.

### CODE REVIEW

File: src/npc.cpp [hunk 1]
```diff
@@ -2045,6 +2045,38 @@ void npc::die( Creature *nkiller )
     place_corpse();
 }
 
+std::string npc_attitude_id( npc_attitude att )
+{
+    static const std::unordered_map<npc_attitude, std::string> npc_attitude_ids = {
```
Comment (kevingranade) [inline src/npc.cpp]: Some compiler versions are unhappy with this, try using a std::map instead.

Comment (ZhilkinSerg) [inline src/npc.cpp]:
```suggestion
    static const std::map<npc_attitude, std::string> npc_attitude_ids = {
```

Comment (ZhilkinSerg) [reply src/npc.cpp]: Let's try, but that's kinda strange - it was a direct copy from some other place in the code and should be working.

Maybe I need to add hash function overload for `npc_attitude`?

```CPP
namespace std
{
template <>
struct hash<npc_attitude> {
    std::size_t operator()( const npc_attitude &k ) const {
        return k;
    }
};
} // namespace std
```

**Comment by kevingranade** [comment]
The reason I suggested std::map instead of adding a hash function is with the numbers of entries you have there, a hash map probably isn't any faster, so no need to invest in it.

**Pull request merged**